dbt_worker/tasks.py

```diff
@@ -265,14 +265,21 @@ def _handle_abort(task_id, p, callback_url):
     # Process is not alive, simply return.
     if not p.is_alive():
         return
-    try:
-        # Try to kill process using SIGINT.
-        # TODO: Send SIGTERM after a timeout-- there is
-        # a bug in core that sometimes makes SIGINT ineffective
-        os.kill(p.pid, signal.SIGINT)
-    except Exception as e:
-        logger.info(str(e))
-
+    # Wait for process to exit or retry killing it with SIGINT
+    retry_timeout = 5  # seconds
```

Comment:
Would this block the endpoint?

Reply:
Yeah good call. I haven't observed it blocking but definitely want you to be able to try on your side to make sure. I added this based on advice from @dichenqiandbt that core has a bug that sometimes causes sigints to time out, but I'm also okay holding off on this addition if it feels like an antipattern prior to actually hitting that bug. Thoughts?

Reply:
I also realized that the sleep was set to .1 instead of 1 -- I'd done this for testing, we don't need to send that many SIGINTs